src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1208,19 +1223,23 @@ private static ExecutorService getExecutorService() {
     }
 
     private PeriodicShardSyncStrategy createPeriodicShardSyncStrategy() {
-        return new PeriodicShardSyncStrategy(createPeriodicShardSyncManager());
+        return new PeriodicShardSyncStrategy(leaderElectedPeriodicShardSyncManager);
     }
 
-    private ShardEndShardSyncStrategy createShardEndShardSyncStrategy(ShardSyncTaskManager shardSyncTaskManager) {
-        return new ShardEndShardSyncStrategy(shardSyncTaskManager);
+    private ShardEndShardSyncStrategy createShardEndShardSyncStrategy() {
+        return new ShardEndShardSyncStrategy(controlServer, leaderElectedPeriodicShardSyncManager);
     }
 
     private LeaderDecider createLeaderDecider() {
         return new DeterministicShuffleShardSyncLeaderDecider(leaseCoordinator.getLeaseManager(),
                 Executors.newSingleThreadScheduledExecutor(), PERIODIC_SHARD_SYNC_MAX_WORKERS_DEFAULT);
     }
 
-    private PeriodicShardSyncManager createPeriodicShardSyncManager() {
+    private PeriodicShardSyncManager getOrCreatePeriodicShardSyncManager(PeriodicShardSyncManager periodicShardSyncManager) {
+        if (periodicShardSyncManager != null) {
+            return periodicShardSyncManager;
+        }
+
         return new PeriodicShardSyncManager(config.getWorkerIdentifier(),
```

Comment:
Actually we need to introduce the mode in which the periodic shard sync will be running. The mode should facilitate running either mandatory shard sync (Periodic) or hashrange validation based shard sync (Shard end). We can add this when we work on incorporating partial hashrange cases as well.

Reply:
Yep I was planning on adding that mode to PeriodicShardSyncManager as part of the partial hashrange story. I'll add a TODO in the meantime.